src/labone/nodetree/node.py

```diff
@@ -87,22 +88,86 @@ def __init__(
             NestedDict[list[list[NormalizedPathSegment]] | dict],
         ] = {}
 
-        paths_as_segments = [split_path(path) for path in self._paths]
+        self._root_prefix: tuple[str, ...] = ()
+        self._paths_as_segments: list[list[NormalizedPathSegment]] = []
+        self.add_nodes_with_info(path_to_info, warnings_active=False)
+
+    def add_nodes_with_info(
+        self,
+        path_to_info: dict[LabOneNodePath, NodeInfoType],
+        *,
+        warnings_active: bool = True,
```

Comment:
issue: This is not necessary. Python already provides context managers to suppress warnings if required. So this function can safely raise a warning.

Reply:
I am not able to find out how to catch the logger statement. It is triggered on each nodetree initialisation however, so catching it would be important

Reply:
https://docs.python.org/3/library/warnings.html#temporarily-suppressing-warnings